src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -466,6 +467,33 @@ RequestContext<object> requestContext
             await requestContext.SendResult(result.First().ToString());
         }
 
+        private async Task HandleGetCommandsRequest(
+            object param,
+            RequestContext<object> requestContext)
+        {
+            var psCommand = new PSCommand();
+            psCommand.AddScript("Get-Command -CommandType Function, Cmdlet, ExternalScript");
+            var result = await this.editorSession.PowerShellContext.ExecuteCommand<PSObject>(psCommand);
+
+            var commandList = new List<PSCommandMessage>();
+
+            if (result != null)
+            {
+                foreach (dynamic c in result)
```

Comment:
I think these are just PSObjects. No need for the dynamic keyword, I think

Reply:
Looking through [the docs](https://docs.microsoft.com/en-au/dotnet/api/system.management.automation.psobject?view=powershellsdk-1.1.0), `PSObject` doesn't have a simple way to access its members, but it does implement [`IDynamicMetaObjectProvider`](https://docs.microsoft.com/en-au/dotnet/api/system.dynamic.idynamicmetaobjectprovider?view=netframework-4.5.1), which looks like its specifically intended for `dynamic` usage. So it looks like `dynamic` is the best (and intended) way to go here.

Reply:
I don't know which way is best, but this is the way I typically use/see most often

```csharp
pso.Properties["PropertyName"].Value;
pso.Methods["MethodName"].Invoke(arg1, arg2);
```

Reply:
Ah, I wondered if that was possible. I imagine using `dynamic` is a syntactic sugar for that. It's possible that there's some overhead on using the `dynamic` way, hooking up the interface, but it's probably minor enough that it comes down to preference

Reply:
I'm going to side with @rkeithhill on this one. Dynamic may be syntactic sugar, but I tried to set it up with `command.Properties["PropertName"].Value` and needed a lot of `.ToString()` and then there's two I haven't resolved yet because they require explicit casting. IMHO it is harder to follow:

```C#
foreach (PSObject command in result)
                {
                    commandList.Add(new PSCommandMessage
                    {
                        Name = command.Properties["Name"].Value.ToString(),
                        ModuleName = command.Properties["ModuleName"].Value.ToString(),
                        Parameters = command.Properties["Parameters"].Value, # This line is missing an explicit cast according to vscode
                        ParameterSets = command.Properties["ParameterSets"].Value, # This line also missing explicit cast.
                        DefaultParameterSet = command.Properties["DefaultParameterSet"].Value.ToString()
                    });
                }
```

@tylerl0706 / @rjmholt are we OK to leave it with dynamic? (I've expanded it from c to command so it's a littler clearer what we're working with.

Reply:
I'm in favour of `dynamic` usage

Reply:
same